core/dataset_manager.py

```diff
@@ -152,69 +160,68 @@ def _create_dataset_mappings(self, mapping_filepath):
 
-        IMPORTANT: NOT FINISHED DEBUGGING, DO NOT USE
+        See _generate_ed_directory docstring for an example of what an ED
+        Directory looks like.
         """
-        filepath = self._raw_filepath
-        self.check_key_length(name)
-        value = {}
-        folders = []
-        for file in os.listdir(filepath):
-            if os.path.isdir(os.path.join(os.path.abspath(filepath), file)):
-                folders.append(file)
-        for folder in folders:
-            folder_dict = {}
-            folder_path = os.path.join(os.path.abspath(filepath), folder)
-            files = os.listdir(folder_path)
-            for file in files:
-                if file[:2] == 'md':
-                    file_path = os.path.join(folder_path, file)
-                    metadata = pd.read_csv(file_path)
-                    folder_dict['md'] = metadata.to_json()
-                else:
-                    file_path = os.path.join(folder_path, file)
-                    dataset = pd.read_csv(file_path)
-                    sample = dataset.sample(frac=0.1)
-                    folder_dict['ds'] = sample.to_json()
-            if 'md' not in folder_dict:
-                raise NoMetadataFoundError(folder)
-            value[folder] = folder_dict
-        receipt = setter(client=self._client, key=name, value=value, port=self._port)
-
-    def post_dataset(self, name):
-        """
-        Post samples of datasets on blockchain with automatically generated
-        metadata under provided name as the key
-
-        IMPORTANT: NOT FINISHED DEBUGGING, DO NOT USE
-        """
-        filepath = self._raw_filepath
-        self.check_key_length(name)
-        value = {}
-        folders = []
-        for file in os.listdir(filepath):
-            if os.path.isdir(os.path.join(os.path.abspath(filepath), file)):
-                folders.append(file)
-        for folder in folders:
-            folder_dict = {}
-            folder_path = os.path.join(os.path.abspath(filepath), folder)
-            file = list(os.listdir(folder_path))[0]
-            file_path = os.path.join(folder_path, file)
-            dataset = pd.read_csv(file_path)
-            md = pd.DataFrame(dataset.describe())
-            sample = dataset.sample(frac=0.1)
-            folder_dict['ds'] = sample.to_json()
-            folder_dict['md'] = md.to_json()
-            value[folder] = folder_dict
-        receipt = setter(client=self._client, key=name, value=value, port=self._port)
+        assert len(key) <= 30, \
+            "Keys for datasets can only be at most 30 characters long."
+        assert self._db_client, \
+            "DB client has not been set. Dataset Manager needs to be configured!"
+        assert self._ipfs_client, \
+            "IPFS client has not been set. Dataset Manager needs to be configured!"
+        
+        ed_directory = self._generate_ed_directory()
+
+        self._db_client.add_labels([key], [self.label])
```

Comment:
Why is add_labels called here? Not too sure why the db_client should be doing something in the method that pushes to the blockchain.

Reply:
Category labels need to be pushed at the same time as ED Directories, since the labels are used to retrieve the directories later.

But you're right, I'll change the method name to be more clear about this.